---
title: 'Q3: Answer'
author: "Steven Shideler"
date: '2022-06-23'
output: html_document
---

#Question 3 Answers:

# Part (A): Generate standard variant files from the gold standards BAM Files from the lab. 

In the past, for prokaryotic variant file generation, I have used the free bcftools from the bioconda 
respository I would use the following commands:

bcftools mpileup -Ob -o <gold_standard.bcf> -f <hg38_ref.fa> <gold_standard.bam> 
bcftools call -vmO z -o <gold_standard.vcf.gz> <gold_standard.bcf>

These commands could / would be added to a bash script or pipeline using snakemake or nextflow for further automation and batch runs. 

However for human genome variant calling I would use the following packages: 

1. Picard tool to remove duplicates (not if high coverage WGS)
2. Bedtools genomecov to determine read coverage (on-target ratio of mapped reads & on-target coverage if Panel Seq or WES)
3. Variant calling packages, Mutect is widely used but is restricted to somatic cell sequences. Strelka or Platypus might be 
better options since they can variant call with somatic or germline cell sequences. It is also important for QC that confidence
scores should be reported. (IF samples are FFPE, due to DNA-crosslinking / potential cytosine deamination further filtering steps
may be needed).

# Part (B):

How do you propose to valide these variants?

I would use the vcf-validator which is a validation suite for VCF files.

https://github.com/EBIvariation/vcf-validator.git

"This suite includes all the checks from the vcftools suite as well as 
additional lexical, syntactic and semantic analysis of the VCF input."

This pipeline not only validates and checks for sytnax errors that might 
have been created during the generation of the vcf files, it has the added 
functionality of finding and fixing simple errors such as the presence of 
duplicates and with the help of a reference file, can also check the correctness 
of the sequences within a vcf file. 
