---
title: 'Q3: Answer'
author: "Steven Shideler"
date: '2022-06-23'
output: html_document
---

#Question 3 Answers:

Part (A):

Generate standard variant files from the gold standards BAM Files from the lab. 

Using free bcftools from the bioconda respository I would use the following commands:

bcftools mpileup -Ob -o <gold_standard.bcf> -f <hg38_ref.fa> <gold_standard.bam> 
bcftools call -vmO z -o <gold_standard.vcf.gz> <gold_standard.bcf>

These commands could / would be added to a bash script or pipeline using snakemake or nextflow for further automation and batch runs. 

Part (B):

How do you propose to valide these variants?

I would use the vcf-validator which is a validation suite for VCF files.

https://github.com/EBIvariation/vcf-validator.git

"This suite includes all the checks from the vcftools suite as well as additional lexical, syntactic and semantic analysis of the VCF input."

This pipeline not only validates and checks for sytnax errors that might have been created during the generation of the vcf files, it has the added functionality of finding and fixing simple errors such as the presence of duplicates and with the help of a reference file, can also check the correctness of the sequences within a vcf file. 